Framework/PythonInterface/mantid/api/src/Exports/Algorithm.cpp

```diff
@@ -76,6 +89,87 @@ void translateCancel(const Algorithm::CancelException &exc) {
   UNUSED_ARG(exc);
   PyErr_SetString(PyExc_KeyboardInterrupt, "");
 }
+
+template <typename T> boost::optional<T> extractArg(ssize_t index, const tuple &args) {
+  if (index < len(args)) {
+    return boost::optional<T>(extract<T>(args[index]));
+  }
+  return boost::none;
+}
+
+template <typename T> void extractKwargs(const dict &kwargs, const std::string &keyName, boost::optional<T> &out) {
+  if (kwargs.has_key(keyName)) {
+    out = boost::optional<T>(extract<T>(kwargs.get(keyName)));
+  }
+}
+
+class SetPropertyVisitor final : public Mantid::PythonInterface::IPyTypeVisitor {
+public:
+  SetPropertyVisitor(Mantid::API::Algorithm_sptr &alg, std::string const &propName)
+      : m_alg(alg), m_propName(propName) {}
+
+  void operator()(bool value) const override { setProp(value); }
+  void operator()(long value) const override { setProp(static_cast<int>(value)); }
```

Comment:
What about if we explicitly pass a workspace as a kwarg, which we can do with the fit algorithm:

```
# create a workspace with a gaussian peak sitting on top of a linear (here flat) background
ws = CreateSampleWorkspace(Function="User Defined", UserDefinedFunction="name=LinearBackground, \
   A0=0.3;name=Gaussian, PeakCentre=5, Height=10, Sigma=0.7", NumBanks=1, BankPixelWidth=1, XMin=0, XMax=10, BinWidth=0.1)

# Setup the data to fit:
workspaceIndex = 0  # the spectrum with which WorkspaceIndex to fit
startX = 1      # specify fitting region
endX = 9      #

# Setup the model, here a Gaussian, to fit to data
tryCentre = '4'   # A start guess on peak centre
sigma = '1'          # A start guess on peak width
height = '8'         # A start guess on peak height
myFunc ='name=Gaussian, Height='+height+', PeakCentre='+tryCentre+', Sigma='+sigma
```

This works, 
```
# normal fit (this works)
fit_output = Fit( Function=myFunc,InputWorkspace=ws, Output='fit')
```

This doesn't work
```
args={"Function": myFunc, "InputWorkspace": ws, "Output": 'fit'}
parent_alg = AlgorithmManager.createUnmanaged('Load')
child_alg = parent_alg.createChildAlgorithm('Fit',0, 0, True, version=1, **args)
child_alg.execute()
```

Reply:
Another good spot. @DavidFair Did we consider how to handle workspaces? I think we might just have missed this...